src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseFileService.cs

```diff
@@ -0,0 +1,258 @@
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.Services
+{
+    public class ReleaseFileService : IReleaseFileService
+    {
+        private readonly ContentDbContext _contentDbContext;
+        private readonly IPersistenceHelper<ContentDbContext> _persistenceHelper;
+        private readonly IBlobStorageService _blobStorageService;
+        private readonly IFileRepository _fileRepository;
+        private readonly IFileUploadsValidatorService _fileUploadsValidatorService;
+        private readonly IReleaseFileRepository _releaseFileRepository;
+        private readonly IUserService _userService;
+
+        public ReleaseFileService(ContentDbContext contentDbContext,
+            IPersistenceHelper<ContentDbContext> persistenceHelper,
+            IBlobStorageService blobStorageService,
+            IFileRepository fileRepository,
+            IFileUploadsValidatorService fileUploadsValidatorService,
+            IReleaseFileRepository releaseFileRepository,
+            IUserService userService)
+        {
+            _contentDbContext = contentDbContext;
+            _persistenceHelper = persistenceHelper;
+            _blobStorageService = blobStorageService;
+            _fileRepository = fileRepository;
+            _fileUploadsValidatorService = fileUploadsValidatorService;
+            _releaseFileRepository = releaseFileRepository;
+            _userService = userService;
+        }
+
+        public async Task<Either<ActionResult, Unit>> Delete(Guid releaseId,
+            Guid id,
+            bool forceDelete = false)
+        {
+            return await Delete(releaseId, new List<Guid>
+            {
+                id
+            }, forceDelete: forceDelete);
+        }
+
+        public async Task<Either<ActionResult, Unit>> Delete(Guid releaseId,
+            IEnumerable<Guid> ids,
+            bool forceDelete = false)
+        {
+            return await _persistenceHelper
+                .CheckEntityExists<Release>(releaseId)
+                .OnSuccess(async release => await _userService.CheckCanUpdateRelease(release, ignoreCheck: forceDelete))
+                .OnSuccess(async release =>
+                    await ids.Select(id => _releaseFileRepository.CheckFileExists(releaseId, id, Ancillary, Chart))
+                        .Aggregate())
+                .OnSuccessVoid(async files =>
+                {
+                    foreach (var file in files)
+                    {
+                        if (await _releaseFileRepository.FileIsLinkedToOtherReleases(releaseId, file.Id))
+                        {
+                            await _releaseFileRepository.Delete(releaseId, file.Id);
+                        }
+                        else
+                        {
+                            await _blobStorageService.DeleteBlob(
+                                PrivateFilesContainerName,
+                                AdminReleasePath(
```

Comment:
Could use the `Path` extension here I think. Probably a good idea to look in other places to replace this with as well.

Reply:
Thanks, yep, replaced 3 occurrences in total 🚀